src/contexts/scrollToTopContext.js

```diff
@@ -0,0 +1,14 @@
+import { useEffect } from "react";
+import { useLocation } from "react-router-dom";
+import { useFilter } from "./filterContext";
+
+export const ScrollToTopProvider = ({ children }) => {
+	const location = useLocation();
+	const { category, rating, sortByPrice, priceRange } = useFilter();
+
+	useEffect(() => {
+		window.scrollTo(0, 0);
+	}, [location, category, rating, sortByPrice, priceRange]);
+
```

Comment:
Suggestion : too many dependencies do check

Reply:
this is needed, as i change the state of  any filter it needs to be triggered.